### PR TITLE
fix(jsx-no-leaked-render): avoid unnecessary negation operators and ternary branches deletion

### LINKED ISSUE #3297: strange fix for jsx-no-leaked-render
**Input File**

```tsx
import { css } from "twin.macro";

export interface sortIndicatorProps {
  direction?: "up" | "down";
  removeFromLayout?: boolean;
  className?: string;
}

export const sortIndicatorMarginLeft = "-1em";

export function SortIndicator({
  direction,
  removeFromLayout = true,
  className,
}: sortIndicatorProps) {
  return (
    <span
      className={className}
      css={[
        css`
          font-size: 0.8em;
          &:first-of-type {
            padding-right: 0.5ch;
          }
        `,
        removeFromLayout &&
          css`
            position: absolute;
            margin-left: ${sortIndicatorMarginLeft};
          `,
      ]}>
      {direction ? (direction === "down" ? "▼" : "▲") : ""}
    </span>
  );
}
```

**output file**

```tsx
import { css } from "twin.macro";

export interface sortIndicatorProps {
  direction?: "up" | "down";
  removeFromLayout?: boolean;
  className?: string;
}

export const sortIndicatorMarginLeft = "-1em";

export function SortIndicator({
  direction,
  removeFromLayout = true,
  className,
}: sortIndicatorProps) {
  return (
    <span
      className={className}
      css={[
        css`
          font-size: 0.8em;
          &:first-of-type {
            padding-right: 0.5ch;
          }
        `,
        removeFromLayout &&
          css`
            position: absolute;
            margin-left: ${sortIndicatorMarginLeft};
          `,
      ]}>
      {/* lol what is this ↓↓↓↓↓ */}
      {!!!!!!!!!!!!!!!!!!!!direction && direction === "down" && "▼"}
    </span>
  );
}
```

eslint rule setup

```js
    "react/jsx-no-leaked-render": [
      // enforce that values are coerced to booleans in jsx conditionals
      "error",
      {
        validStrategies: ["coerce"], // coerce to boolean e.g. `<div>{count && title}</div>` becomes `<div>{!!count && title}</div>`
      },
    ],
```

### PR DESCRIPTION
**Pull request by Belco90**
Closes #3297
Relates to #3292

This fixes:
- adding unnecessary negation operators
- delete valid ternary branches

### CODE REVIEW

File: tests/lib/rules/jsx-no-leaked-render.js [hunk 1]
```diff
@@ -124,6 +124,57 @@ ruleTester.run('jsx-no-leaked-render', rule, {
         }
       `,
     },
+    {
+      options: [{ validStrategies: ['coerce'] }],
+      code: `
```
Comment (ljharb) [inline tests/lib/rules/jsx-no-leaked-render.js:129]: let's have code come before options, and let's be sure to include `output` so the fixer is tested

**Review by ljharb — CHANGES_REQUESTED** [review]

File: tests/lib/rules/jsx-no-leaked-render.js (hunk 1 above)
Comment (Belco90) [reply tests/lib/rules/jsx-no-leaked-render.js:129]: >  let's be sure to include output so the fixer is tested

How? This is a valid case.

Comment (ljharb) [reply tests/lib/rules/jsx-no-leaked-render.js]: oh duh, sorry.

**Review by ljharb — APPROVED** [review]
assuming tests pass

**Comment by pke** [comment]
Any plans when this will be released?

**Comment by ljharb** [comment]
Nope, we don’t plan release dates, they just happen when we have time.

**Pull request merged**